src/analyses/goto_check.cpp

```diff
@@ -2089,7 +2091,7 @@ void goto_checkt::goto_check(
           goto_programt::targett t =
             new_code.add(goto_programt::make_assignment(
               std::move(lhs), std::move(rhs), i.source_location));
-          t->code.add_source_location()=i.source_location;
+          t->code_nonconst().add_source_location() = i.source_location;
```

Comment:
This line should be removed and `goto_programt::make_assignment` should be fixed so as to invoke `code_assignt(std::move(lhs), std::move(rhs), l)` (i.e., adding the `, l`).